### PR TITLE
fix(v2): final empty classes, internal base classes

### PR DESCRIPTION
**Pull request by noahdietz**
Make the `v2` empty "service" clients `final` and the `v2` `BaseClient` classes `@internal`.

Fixes http://b/271286629

### CODE REVIEW

**Review by bshaffer — APPROVED** [review]

File: tests/Integration/goldens/iam/src/V1/Client/BaseClient/IAMPolicyBaseClient.php [hunk 1]
```diff
@@ -68,6 +68,8 @@
  * This class provides the ability to make remote calls to the backing service through method
  * calls that map to API methods.
  *
+ * @internal
```
Comment (bshaffer) [inline tests/Integration/goldens/iam/src/V1/Client/BaseClient/IAMPolicyBaseClient.php:71]: Hmm, I just had a thought. I think if we label something as internal it's skipped by the breaking change detector :-(

But i don't know another way we can say "don't use this class directly" that wouldn't affect the check.

Comment (bshaffer) [reply tests/Integration/goldens/iam/src/V1/Client/BaseClient/IAMPolicyBaseClient.php:71]: Hopefully it's smart enough to see the breaking changes in the non-internal surfaces.  I'll have to test it out

Comment (noahdietz) [reply tests/Integration/goldens/iam/src/V1/Client/BaseClient/IAMPolicyBaseClient.php:71]: I wonder if the Empty Client extending the BaseClient would give the BCD a non-internal surface to vet...

Comment (bshaffer) [reply tests/Integration/goldens/iam/src/V1/Client/BaseClient/IAMPolicyBaseClient.php:71]: I just checked, and the BC breaking client throws an error if it finds a non-internal class extending that parent class!

So my fears are invalid, and everything is great.

Comment (noahdietz) [reply tests/Integration/goldens/iam/src/V1/Client/BaseClient/IAMPolicyBaseClient.php:71]: Woo! thanks for raising and for checking :)

**Pull request merged**